Tasks/VsTest/settingshelper.ts

```diff
@@ -55,89 +52,89 @@ const runSettingsTemplate = `<?xml version=\"1.0\" encoding=\"utf-8\"?>
+            tl.warning(tl.loc('testSettingPropertiesNotSupported'))
         }
-    }
 
-    if (isParallelRun) {
-        if (settingsExt === testSettingsExtension) {
-            tl.warning(tl.loc('RunInParallelNotSupported'));
-        } else if (settingsExt === runSettingsExtension) {
-            tl.debug('Enabling run in parallel by editing given runsettings.');
-            result = setupRunSettingsWithRunInParallel(result);
-        } else {
-            tl.debug('Enabling run in parallel by creating new runsettings.');
-            settingsExt = runSettingsExtension;
-            result = await CreateSettings(runSettingsForParallel);
+        if (overrideParametersString) {
+            if (settingsExt === runSettingsExtension || settingsExt === testSettingsExtension) {
+                result = updateSettingsWithParameters(result, overrideParametersString);
+            } else {
+                tl.warning(tl.loc('overrideNotSupported'));
+            }
         }
-    }
 
-    if (videoCollector) {
-        //Enable video collector only in test settings.
-        let videoCollectorNode = null;
-        parser.parseString(videoDataCollectorTemplate, function (err, data) {
-            if (err) {
-                defer.reject(err);
+        if (isParallelRun) {
+            if (settingsExt === testSettingsExtension) {
+                tl.warning(tl.loc('RunInParallelNotSupported'));
+            } else if (settingsExt === runSettingsExtension) {
+                tl.debug('Enabling run in parallel by editing given runsettings.');
+                result = setupRunSettingsWithRunInParallel(result);
+            } else {
+                tl.debug('Enabling run in parallel by creating new runsettings.');
+                settingsExt = runSettingsExtension;
+                result = CreateSettings(runSettingsForParallel);
             }
-            videoCollectorNode = data;
-        });
-        if (settingsExt === testSettingsExtension) {
-            tl.debug('Enabling video data collector by editing given testsettings.')
-            result = updateTestSettingsWithDataCollector(result, videoCollectorFriendlyName, videoCollectorNode);
-        } else if (settingsExt === runSettingsExtension) {
-            tl.warning(tl.loc('VideoCollectorNotSupportedWithRunSettings'));
-        } else {
-            tl.debug('Enabling video data collection by creating new test settings.')
-            settingsExt = testSettingsExtension;
-            result = await CreateSettings(testSettingsTemplate);
-            result = updateTestSettingsWithDataCollector(result, videoCollectorFriendlyName, videoCollectorNode)
         }
-    }
 
-    if (tiaConfig.tiaEnabled && !tiaConfig.disableEnablingDataCollector) {
-        let testImpactCollectorNode = null;
-        parser.parseString(testImpactDataCollectorTemplate, function (err, data) {
-            if (err) {
-                defer.reject(err);
+
+        if (videoCollector) {
```

Comment:
video collector is hard coded false where this function is called. This code will never execute. What was the intention here?

Reply:
@RanjanarMS @nigurr